src/Npgsql/Internal/Converters/Geometric/PathConverter.cs

```diff
@@ -0,0 +1,68 @@
+using System.Diagnostics;
+using System.Threading;
+using System.Threading.Tasks;
+using NpgsqlTypes;
+
+// ReSharper disable once CheckNamespace
+namespace Npgsql.Internal.Converters;
+
+public class PathConverter : PgStreamingConverter<NpgsqlPath>
+{
+    public override NpgsqlPath Read(PgReader reader)
+        => Read(async: false, reader, CancellationToken.None).GetAwaiter().GetResult();
+
+    public override ValueTask<NpgsqlPath> ReadAsync(PgReader reader, CancellationToken cancellationToken = default)
+        => Read(async: true, reader, cancellationToken);
+
+    async ValueTask<NpgsqlPath> Read(bool async, PgReader reader, CancellationToken cancellationToken)
+    {
+        if (reader.Remaining < sizeof(byte) + sizeof(int))
+            await reader.BufferData(async, sizeof(byte) + sizeof(int), cancellationToken).ConfigureAwait(false);
+
+        var open = reader.ReadByte() switch
+        {
+            1 => false,
+            0 => true,
+            _ => throw new UnreachableException("Error decoding binary geometric path: bad open byte")
+        };
+
+        var numPoints = reader.ReadInt32();
+        var result = new NpgsqlPath(numPoints, open);
+
+        for (var i = 0; i < numPoints; i++)
+        {
+            if (reader.Remaining < sizeof(double) * 2)
```

Comment:
Just to make sure, ShouldBuffer also exists on reader. It additionally checks against the max buffer size etc. 

Though I'm completely fine with checking Remaining for these kind of constant sizes.

Reply:
Oh, missed that - ShouldBuffer does look nicer. Will change.